codyze-v2/build.gradle.kts

```diff
@@ -146,61 +107,112 @@ dependencies {
     testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.9.1")
 }
 
+/*
+ * General project configuration
+ */
+
+group = "de.fraunhofer.aisec"
+
+/*
+ * Plugin configuration block
+ */
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
 application {
     mainClass.set("de.fraunhofer.aisec.codyze.Main")
 }
 
-tasks.named<Test>("test") {
-    useJUnitPlatform()
-    maxHeapSize = "6000m"
+jsonSchema2Pojo {
+    source.setFrom("src/main/resources/json")
+    targetPackage.set("de.fraunhofer.aisec.codyze.sarif.schema")
+    removeOldOutput.set(true)
 }
 
+spotless {
+    ratchetFrom("origin/main")
+    java {
+        importOrder()
+        removeUnusedImports()
+        googleJavaFormat()
+        // exclude generated files
+        targetExclude("build/generated/**/*.java")
+    }
+    kotlin {
+        ktfmt().kotlinlangStyle()
+    }
+}
+
+downloadLicenses {
+    includeProjectDependencies = true
+    dependencyConfiguration = "compileClasspath"
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            from(components["java"])
+        }
+    }
+}
+
+/*
+ * Task configuration block
+ */
+
 // Add generated sources from annotation processor to source sets so they can be picked up
 sourceSets.configureEach {
     tasks.named<JavaCompile>(compileJavaTaskName) {
         java.srcDir(options.generatedSourceOutputDirectory.get())
     }
 }
 
+val compileKotlin: KotlinCompile by tasks
+compileKotlin.kotlinOptions {
+    jvmTarget = "11"
+}
+
+val compileTestKotlin: KotlinCompile by tasks
+compileTestKotlin.kotlinOptions {
+    jvmTarget = "11"
+}
+
 tasks.withType<KotlinCompile>().configureEach {
     kotlinOptions {
         freeCompilerArgs = freeCompilerArgs + "-opt-in=kotlin.RequiresOptIn"
     }
 }
 
+// state that JSON schema parser must run before compiling Kotlin
+tasks.named("compileKotlin") {
+    dependsOn(":generateJsonSchema2DataClass")
```

Comment:
It's probably not actually necessary to trash and re-generate the SARIF schema files on every compilation since the schema rarely changes.
(But removing this would mean that the gradle command has to be run manually before compiling for the first time or when changing the schema.)